Sources/KavaPlugin+AnalyticsPluginProtocol.swift

```diff
@@ -0,0 +1,218 @@
+                    PKLog.debug("pause event: \(event)")
+                    strongSelf.kavaData.isPaused = true
+
+                    strongSelf.sendAnalyticsEvent(event: KavaEventType.pause)
+                    strongSelf.reportView()
+                    strongSelf.stopViewTimer()
+                })
+                
+            case let e where e.self == PlayerEvent.playing:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("playing event: \(event)")
+                    if strongSelf.isFirstPlay {
+                        strongSelf.isFirstPlay = false
+                        strongSelf.sendAnalyticsEvent(event: KavaEventType.play)
+                    } else if strongSelf.kavaData.isPaused {
+                        strongSelf.sendAnalyticsEvent(event: KavaEventType.resume)
+                    }
+                    
+                    strongSelf.kavaData.isPaused = false
+                    strongSelf.setupViewTimer()
+                })
+                
+            case let e where e.self == PlayerEvent.seeking:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    PKLog.debug("seeking event: \(event)")
+                    
+                    guard let strongSelf = self else { return }
+                    
+                    if let seekPosition = event.targetSeekPosition {
+                        strongSelf.kavaData.targetSeekPosition = Double(truncating: seekPosition)
+                    }
+                    
+                    strongSelf.sendAnalyticsEvent(event: KavaEventType.seek)
+                })
+                
+            case let e where e.self == PlayerEvent.sourceSelected:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("sourceSelected event: \(event)")
+                    strongSelf.kavaData.selectedSource = event.mediaSource
+                    strongSelf.updateDeliveryType(mediaFormat: (strongSelf.kavaData.selectedSource?.mediaFormat)!)
+                    // Reset flags when source was changed
+                    strongSelf.resetPlayerFlags()
+                })
+                
+            case let e where e.self == PlayerEvent.ended:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("ended event: \(event)")
+                    
+                    strongSelf.sendPercentageReachedEvent(percentage: 100)
+                    strongSelf.reportView()
+                    strongSelf.stopViewTimer()
+                    strongSelf.resetPlayerFlags()
```

Comment:
not sure `ended` is a good place to reset state - what do you get on replay of same media? better have a reset in one deterministic place(if applicable of course, not always the case)